### ISSUE TITLE
How use lsp with two editor Instance?

### ISSUE THREAD

**Issue by FunctionRun** [TypeFox/monaco-languageclient#227]
In single page, have two code editor

```
const editor 1 = monaco.editor.create(document.querySelector('div1'), ...)

const editor 2 = monaco.editor.create(document.querySelector('div2'), ...)

MonacoServices.install(this.editor1, { rootUri: '' });
MonacoServices.install(this.editor2, { rootUri: '' });
```
Now, not support install two editor , any idea?

**Comment by CGNonofr** [comment]
You need to install the services only once, see https://github.com/TypeFox/monaco-languageclient/blob/10f66f4ccfa41c615733071b32e174003f55016f/client/src/monaco-services.ts

Unfortunately, the editor is required for `MonacoCommands` service, so you'll need to implement a multi-editor version of `Commands` (https://github.com/TypeFox/monaco-languageclient/blob/10f66f4ccfa41c615733071b32e174003f55016f/client/src/monaco-commands.ts)

**Comment by FunctionRun** [comment]
> You need to install the services only once, see https://github.com/TypeFox/monaco-languageclient/blob/10f66f4ccfa41c615733071b32e174003f55016f/client/src/monaco-services.ts
> 
> Unfortunately, the editor is required for `MonacoCommands` service, so you'll need to implement a multi-editor version of `Commands` (https://github.com/TypeFox/monaco-languageclient/blob/10f66f4ccfa41c615733071b32e174003f55016f/client/src/monaco-commands.ts)

If I hava one language client, it's ok . But two language client  for different language server

**Comment by CGNonofr** [comment]
I'm having some difficulty understanding, is this another issue you are addressing now?

I don't think there is any issue having multiple language clients at the same time.

**Comment by FunctionRun** [comment]
> I'm having some difficulty understanding, is this another issue you are addressing now?
> 
> I don't think there is any issue having multiple language clients at the same time.

The same like this: https://www.gitmemory.com/issue/TypeFox/monaco-languageclient/94/488968451

**Comment by CGNonofr** [comment]
Is this a third issue?

I don't think the protocol allow to do such things, I don't see how it could.

however, some implementations allow via socket to connect more than one client.
